X9.146/README.md

```diff
@@ -0,0 +1,406 @@
+# X9.146 Examples
+
+This README file explains how to setup various demos for showing our X9.146
+features in action. X9.146 is a specification of a certificate format that
+allows for dual public keys and signatures in a single certificate.
+
+Traditionally, there are only public key, signature algoirthm specifier and
+signature value. These are known as the native elements. The X9.146 scheme also
+allows for additional alternative public key, signature algorithm specifier and
+signature value as optional X.509 certificate extensions.
+
+The X9.146 specification also specifies how to use these certificates in TLS
+1.3. In the ClientHello message, a CKS extension is added. This extension
+specifies the ability and preference for which signature(s) is/are sent in the
+CertificateVerify message. The presence of the value specifies ability; the
+order of the values specifies preference. The following values are defined: 
+
+- NATIVE 0x01
+- ALTERNATIVE 0x02
+- BOTH 0x03
+- EXTERNAL 0x04 (not supported)
+
+The ServerHello message would have the extension and it would only have a single
+value which would be one of the ones in the list sent over by the client.  That
+is going to specify what is sent in the CertificateVerify message. BOTH is simply the concatenation of the native and alternative signatures; native first.
+
+## Post-Quantum
+
+Tested with these wolfSSL build options:
+
+```sh
+./autogen.sh  # If cloned from GitHub
+./configure --enable-x9-146 --with-liboqs --enable-debug
+make
+sudo make install
+sudo ldconfig # required on some targets
+```
+
+In the directory where this README.md file is found, clean up previous build
+products and certificates and then build the applications.
+
+```sh
+make clean all
+```
+NOTE: `clean` removes certificates and keys in this directory.
+
+### What to Expect
+
+There will be a lot of debug output going to stderr. On the client side, during
+the call to `DoTls13Certificate()`, please search for the following messages to
+confirm that the alternative signature was verified:
+
+```
+Alternative signature has been verified!
+Verified Peer's cert
+```
+
+These debug messages indicate that the client has verified the alternative
+post-quantum certificate chain. The second message indicates that normal
+verification was also successful.
+
+On the client side, during the call to `DoTls13CertificateVerify()` look for
+messages that indicate both conventional and post-quantum verification:
+
+For example, if you are doing ECDSA with Falcon, you will see the following:
+
+```
+Doing ECC peer cert verify
+wolfSSL Entering EccVerify
+wolfSSL Leaving EccVerify, return 0
+Doing Falcon peer cert verify
+wolfSSL Leaving DoTls13CertificateVerify, return 0
+```
+
+### ECDSA Demos
+
+#### P-256 and Dilithium Level 2 Demo
+Generate the various conventional keys; the post-quantum key are pre-generated:
+
+```sh
+openssl genpkey -algorithm ec -pkeyopt ec_paramgen_curve:P-256 -out ca-key.der -outform der

```

Comment:
wolfCLU can do this!!

Reply:
Yes, but at wolf clu can't do the pkey command for pq algos.  I'd  switch to  wolfCLUif if the user could do it all with wolfCLU,